File: backend/app/models/inquiry.py
"""
1대1 문의 및 요청사항 SQLAlchemy 데이터 모델 (한글 주석 적용)
"""
from sqlalchemy import Column, Integer, String, Text, DateTime
from app.core.database import Base
from app.utils.datetime_kst import utc_now

class Inquiry(Base):
    __tablename__ = "inquiries"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    user_email = Column(String(255), nullable=False)
    store_name = Column(String(255), nullable=True, default="포슬카페")
    category = Column(String(100), nullable=False)
    title = Column(String(255), nullable=False)
    content = Column(Text, nullable=False)
    status = Column(String(50), default="pending")  # 'pending' or 'answered'
    answer = Column(Text, nullable=True)
    # 저장은 UTC(naive)로 통일한다 — 화면에 뿌릴 때 utils.datetime_kst.fmt_kst로 KST 변환.
    # 여기서 바로 KST를 넣으면 이미 쌓인 UTC 행들과 섞여 목록의 시각 기준이 둘이 된다.
    created_at = Column(DateTime, default=utc_now)
    answered_at = Column(DateTime, nullable=True)

File: frontend/src/lib/api/inquiry.ts
// 백엔드 B 담당 — 1대1 문의 API (백엔드 /api/v1/inquiries 연동)
// 관리자 답변 도착 감지(AlertsWatcher)와 설정 화면 '나의 문의 내역'이 공용으로 사용한다.
import { apiFetch } from './client';

export type Inquiry = {
  id: number;
  user_email: string;
  store_name: string;
  category: string;
  title: string;
  content: string;
  status: 'pending' | 'answered';
  answer: string | null;
  date: string;
};

/** 내 문의 내역 조회 — 서버가 토큰 주인의 문의만 최신순으로 돌려준다.
 *
 *  예전엔 `?user_email=`에 이메일만 붙이면 인증 없이 남의 문의 전문과 관리자 답변까지
 *  읽을 수 있었다. 이제 이메일을 보내지 않고, 누구 것인지는 토큰이 정한다.
 */
export function listMyInquiries(token: string): Promise<Inquiry[]> {
  return apiFetch('/api/v1/inquiries', {
    headers: { Authorization: `Bearer ${token}` },
  });
}
